### PR TITLE
Replace TimedApi with Micrometer's Timed annotation

### PR DESCRIPTION
**Pull request by adutra**
This change replaces the custom `@TimedApi` annotation with Micrometer's `@Timed` annotation. This should facilitate future integration with Quarkus. For the Dropwizard runtime, this change should be a no-op.

As a nice side-effect, polaris-core does not depend on micrometer anymore.

### CODE REVIEW

File: service/common/build.gradle.kts [hunk 1]
```diff
@@ -83,6 +83,9 @@ dependencies {
 
   implementation(platform(libs.azuresdk.bom))
   implementation("com.azure:azure-core")
+
+  implementation(platform(libs.micrometer.bom))
```
Comment (snazy) [inline service/common/build.gradle.kts:87]: Nit: the platform can be compileOnly as well
```suggestion
  compileOnly(platform(libs.micrometer.bom))
```

**Review by snazy — COMMENTED** [review]
Nice improvement!
I'm really looking forward to the Quarkus stuff, which doesn't require us to implement these infrastructure things ourselves.

File: dropwizard/service/src/main/java/org/apache/polaris/service/dropwizard/TimedApplicationEventListener.java [hunk 2]
```diff
@@ -18,28 +18,28 @@
...
 
 /**
  * An ApplicationEventListener that supports timing and error counting of Jersey resource methods
- * annotated by {@link TimedApi}. It uses the {@link PolarisMetricRegistry} for metric collection
- * and properly times the resource on success and increments the error counter on failure.
+ * annotated by {@link Timed}. It uses the {@link PolarisMetricRegistry} for metric collection and
+ * properly times the resource on success and increments the error counter on failure.
  */
 @Provider
 public class TimedApplicationEventListener implements ApplicationEventListener {
```
[the remark this answers is not in the document]
Comment (adutra) [reply dropwizard/service/src/main/java/org/apache/polaris/service/dropwizard/TimedApplicationEventListener.java:45]: The listener is only for Dropwizard, Quarkus doesn't need it.

[the remark this answers is not in the document]
Comment (jbonofre) [reply dropwizard/service/src/main/java/org/apache/polaris/service/dropwizard/TimedApplicationEventListener.java:45]: That's exactly my point (remember I removed it in the first Quarkus PoC PR).

**Review by jbonofre — APPROVED** [review]

**Pull request merged**